access_error.go

```diff
@@ -38,12 +38,16 @@ func (f *Fosite) writeJsonError(rw http.ResponseWriter, err error) {
 
 	rfcerr := ErrorToRFC6749Error(err)
 	if !f.SendDebugMessagesToClients {
-		rfcerr.Debug = ""
+		rfcerr = rfcerr.Sanitize()
 	}
 
-	js, err := json.Marshal(rfcerr)
+	js, err := json.MarshalIndent(rfcerr, "", "\t")
 	if err != nil {
-		http.Error(rw, fmt.Sprintf(`{"error": "%s"}`, err.Error()), http.StatusInternalServerError)
+		if !f.SendDebugMessagesToClients {
+			http.Error(rw, fmt.Sprintf(`{\n\t"error": "server_error",\n\t"error_description": "%s"\n}`, err.Error()), http.StatusInternalServerError)
+		} else {
+			http.Error(rw, `{\n\t"error": "server_error"\n}`, http.StatusInternalServerError)
```

Comment:
So this was bad and I fixed it:
* It was not a proper OAuth response (`error` should be one of error constants, not exception string).
* It might leak sensitive data.

Reply:
```suggestion
			http.Error(rw, `{"error":"server_error"}`, http.StatusInternalServerError)
```